libcasr/src/report.rs

```diff
@@ -774,3 +774,230 @@ pub fn cluster_reports(casreps: &[CrashReport]) -> Result<Vec<u32>> {
+                "    #2 0x421f13 in fuzzer::Fuzzer::CrashCallback() /llvm-project-llvmorg-14.0.6/compiler-rt/lib/fuzzer/FuzzerLoop.cpp:233:3".to_string(),
+                "    #3 0x7ffff7dc641f  (/lib/x86_64-linux-gnu/libpthread.so.0+0x1441f) (BuildId: 7b4536f41cdaa5888408e82d0836e33dcf436466)".to_string(),
+                "    #4 0x4c4540 in runtime.raise.abi0 runtime/sys_linux_amd64.s:158".to_string(),
+                "".to_string(),
+                "NOTE: libFuzzer has rudimentary signal handlers.".to_string(),
+                "      Combine libFuzzer with AddressSanitizer or similar for better crash reports.".to_string(),
+                "SUMMARY: libFuzzer: deadly signal".to_string(),
+                "".to_string(),
+            ],
+            rust_report: vec![
+                "Running: ./artifacts/fuzz_target_1/crash-da39a3ee5e6b4b0d3255bfef95601890afd80709".to_string(),
+"thread '<unnamed>' panicked at fuzz_targets/fuzz_target_1.rs:6:9:".to_string(),
+"index out of bounds: the len is 0 but the index is 10".to_string(),
+"stack backtrace:".to_string(),
+" 0:     0x557e259f793c - std::backtrace_rs::backtrace::libunwind::trace::h7d5a50c97105e9c9".to_string(),
+"                             at /rustc/59a8294849358a878a72358aa6d5fe5b9d312867/library/std/src/../../backtrace/src/backtrace/libunwind.rs:93:5".to_string(),
+" 1:     0x557e259f793c - std::backtrace_rs::backtrace::trace_unsynchronized::hf283bd0ba71b8b19".to_string(),
+"                             at /rustc/59a8294849358a878a72358aa6d5fe5b9d312867/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5".to_string(),
+"                             at /home/toka/.cargo/registry/src/index.crates.io-6f17d22bba15001f/libfuzzer-sys-0.4.7/libfuzzer/FuzzerMain.cpp:20:30".to_string(),
+"2:     0x7ffbc33c6d90 - __libc_start_call_main".to_string(),
+"                            at ./csu/../sysdeps/nptl/libc_start_call_main.h:58:16".to_string(),
+"3:     0x7ffbc33c6e40 - __libc_start_main_impl".to_string(),
+"                            at ./csu/../csu/libc-start.c:392:3".to_string(),
+"4:     0x557e25886205 - _start".to_string(),
+"5:                0x0 - <unknown>".to_string(),
+            ],
+            source: vec![
+                "--->83             return utf16_to_utf8(std::u16string(name_array.begin(),".to_string(),
+            ],
+            execution_class: ExecutionClass {
+                severity: "NOT_EXPLOITABLE".to_string(),
+                short_description: "SourceAv".to_string(),
+                description: "Access violation on source operand".to_string(),
+                explanation: "The target crashed on an access violation at an address matching the source operand of the current instruction. This likely indicates a read access violation.".to_string(),
+            },
+            pid: 16476,
+        };
+        assert_eq!(
+            159,
+            format!("{report}").split('\n').collect::<Vec<&str>>().len()
```

Comment:
We need to check not only size, but content too (maybe for some critical parts only).

Reply:
Done